src/app/qgssnappingwidget.cpp

```diff
@@ -197,18 +197,13 @@ QgsSnappingWidget::QgsSnappingWidget( QgsProject *project, QgsMapCanvas *canvas,
     QToolButton *topoButton = new QToolButton();
     topoButton->addAction( mTopologicalEditingAction );
     topoButton->setDefaultAction( mTopologicalEditingAction );
-    if ( mDisplayMode == Widget )
-    {
-      topoButton->setToolButtonStyle( Qt::ToolButtonTextBesideIcon );
-    }
+    topoButton->setToolButtonStyle( Qt::ToolButtonTextBesideIcon );
```

Comment:
why removing the test on the displayMode?
I think this will lead to having text in the toolbar (did not test though)

Reply:
Aren't we already in the widget mode here (https://github.com/qgis/QGIS/blob/master/src/app/qgssnappingwidget.cpp#L182)?

Reply:
you're correct. It remains from the time where there was a status bar mode too!
LGTM
thanks for handling this